### ISSUE TITLE
Refactor Files under Common/src/System/CodeDom

### ISSUE THREAD

**Issue by shmao** [dotnet/corefx#15154]
Refactor the files under src/Common/src/System/CodeDom and remove the ifdefs in the files. See the following comments copied from https://github.com/dotnet/corefx/pull/15054 for more details:

>I took a quick look. It looks like the serialization lib is using this CodeDom support for parsing the type arguments out of a type name string? Could we just factor out that parsing into its own helper that's then used by both CodeDom and the serialization lib, rather than sharing these whole files and using these ifdefs to control namespace and visibility?

**Comment by karelz** [comment]
@shmao please copy the reason from the comment in the PR - it is very hard to follow these links, especially after some time. Thanks!

**Comment by shmao** [comment]
> @shmao please copy the reason from the comment in the PR - it is very hard to follow these links, especially after some time. Thanks!

Sure. I edited the issue description.

**Comment by karelz** [comment]
Thank you!

**Comment by Priya91** [comment]
The ask is to separate out code that's used by Data contract serialization from the codedom files into a common shared code location, instead of moving the whole codedom files. Hence putting the work under serialization area path.
